runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/AvroByteReader.java

```diff
@@ -58,17 +56,17 @@
   final Coder<T> coder;
   private final Schema schema = Schema.create(Schema.Type.BYTES);
 
-  public AvroByteReader(
+  AvroByteReader(
       String filename,
       long startPosition,
       long endPosition,
       Coder<T> coder,
       PipelineOptions options) {
-    this.filename = checkNotNull(filename, "filename");
```

Comment:
Why remove these precondition checks?

Since `@Nonnull` is not enabled by default for the project, is it worth adding for these individual parameters?